### PR TITLE
Remove samstoflag subject aspect

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

**Comment by shriramshankar** [comment]
Once I clean up `nouns/subjects`,  the code coverage will be back to normal.

File: db/model/aspect.js [hunk 1]
```diff
@@ -179,21 +177,27 @@ module.exports = function aspect(seq, dataTypes) {
     },
     hooks: {
 
+      /**
+       * TODO:
```
Comment (iamigo) [inline db/model/aspect.js]: "TODO"? Is there user story for this? Or are you doing this in subsequent PR for this same story?

File: db/model/subject.js [hunk 2]
```diff
@@ -235,6 +187,14 @@ module.exports = function subject(seq, dataTypes) {
     },
     hooks: {
 
+      /**
+       * TODO:
```
Comment (iamigo) [inline db/model/subject.js]: same question as above

File: db/model/subject.js [hunk 3]
```diff
@@ -262,23 +222,21 @@ module.exports = function subject(seq, dataTypes) {
        *  record
        */
       afterCreate(inst /* , opts */) {
+        /*
+         * add entry to the subjectStore in redis only if the subject
+         * is published and the sampleStoreFeature is enabled
```
Comment (iamigo) [inline db/model/subject.js]: really? don't *all* subjects get added to sample store when sampleStoreFeature is enabled, even unpublished ones?

File: db/model/subject.js [hunk 4]
```diff
@@ -435,9 +383,7 @@ module.exports = function subject(seq, dataTypes) {
               common.publishChange(inst, eventName.del);
 
               // if cache is on, remove reference to subjects in the cache
```
Comment (iamigo) [inline db/model/subject.js]: Comment is incorrect here now

File: db/model/subject.js (hunk 3 above)
Comment (shriramshankar) [reply db/model/subject.js]: I did not add this comment; but yes, we are adding subjects to the sampleStore only when it is published. The aspects are added no matter what. I have a note to create stories to fix issues like this.

File: db/model/aspect.js (hunk 1 above)
Comment (shriramshankar) [reply db/model/aspect.js]: I will be creating stories to address the TODOs.

**Comment by shriramshankar** [comment]
My next set of changes that are based on this PR is ready. @jgraff2 Can you please review this? 
@iamigo   I saw your comments, is there something you want me to change.?

**Comment by shriramshankar** [comment]
I also deployed this branch to staging. Everything works as expected.

File: tests/db/model/aspect/utils.js [hunk 5]
```diff
@@ -49,12 +53,16 @@ module.exports = {
   },
 
   forceDelete(done) {
-    tu.forceDelete(tu.db.Sample, testStartTime)
-    .then(() => tu.forceDelete(tu.db.Aspect, testStartTime))
+    Promise.join(rcli.flushallAsync(),
+    tu.forceDelete(tu.db.Aspect, testStartTime)
```
Comment (jgraff2) [inline tests/db/model/aspect/utils.js]: indentation

File: tests/db/model/subject/utils.js [hunk 6]
```diff
@@ -24,11 +26,13 @@ const subjectPrototype = {
 
 module.exports = {
   forceDelete(done) {
+    Promise.join(rcli.flushallAsync(),
     tu.forceDelete(tu.db.Sample, testStartTime)
```
Comment (jgraff2) [inline tests/db/model/subject/utils.js]: indentation

**Comment by shriramshankar** [comment]
Will take care of the comments in the next PR.

**Pull request merged**